pkg/metrics/metrics.go

```diff
@@ -124,7 +128,7 @@ func New(ctx context.Context, appID, version string, opt ...Option) (*Client, er
 	// Use ParseRequestURI over Parse because Parse validation is more loose and will accept
 	// things such as relative paths without a host.
 	if _, err := url.ParseRequestURI(c.ServerURL); err != nil {
-		return noopClient(), fmt.Errorf("failed to parse server url: %w", err)
+		return noopClient(), fmt.Errorf("failed to parse server URL: %w", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("failed to parse server URL: %w", err)
```

Callers should not assume there's a valid client if they get an error.

Reply:
I was intentionally braking the convention a bit since there isn't really anything a caller can do with the error, so I had it just return a noop client in any error case so the caller could ignore errors and not need to worry about handling nil.

```go
metricsClent, _ := metrics.New()
```

If bucking conventions seems like a bad idea, I could change to return nil and make `NoopClient()` public.

```go
metricsClent, err := metrics.New()
if err != nil {
  metricsClient = NoopClient()
}
```

Reply:
Why would you want a bad client? Wouldn't you just bail (return) without a client?

Reply:
I see - you want the business logic to always publish a metric?

Reply:
I figured it would be easier. If metrics are not ergonomic people will avoid using them

Reply:
In that case:

```go
metricsClent, err := metrics.New()
if err != nil {
  metricsClient = NoopClient()
}
```

is the most idiomatic

Reply:
https://github.com/abcxyz/abc-updater/pull/61